internal/generator/vector/output/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,159 @@
+package cloudwatch
+
+import (
+	"fmt"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	. "github.com/openshift/cluster-logging-operator/internal/generator"
+	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+	corev1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+type CloudWatch struct {
+	ID_Key      string
+	Desc        string
+	ComponentID string
+	Inputs      string
+	Endpoint    string
+	Region      string
+	//if Endpoint provided then it overrides  Region         string
+	Host           string
+	GroupBy        string
+	LogGroupPrefix string
+}
+
+func (e CloudWatch) Name() string {
+	return "cloudwatchTemplate"
+}
+
+func (e CloudWatch) Template() string {
+
+	return `{{define "` + e.Name() + `" -}}
+
+[sinks.{{.ComponentID}}]
+type = "aws_cloudwatch_logs"
+inputs = {{.Inputs}}
+region = "{{.Region}}"
+create_missing_group = true
+create_missing_stream = true
+compression = "none"
+encoding.codec = "json"
+request.concurrency = 2
+group_name = "{{"{{ Cw_groupName }}"}}"
+stream_name = "{{"{{ Cw_streamName }}"}}"
+{{- end}}
+`
+}
+
+func AddCWGroupNameAndStreamName(groupbytype string, logGroupPrefix string, id string, inputs []string) Element {
+	return RemapCW{
+		Desc:           "Adding group_name and stream_name field",
+		ComponentID:    id,
+		Inputs:         helpers.MakeInputs(inputs...),
+		GroupBy:        groupbytype,
+		LogGroupPrefix: logGroupPrefix,
+		VRL: strings.TrimSpace(`
+
+   if (.file != null) {
+     filepath = replace!(.file, "/", ".")
+     .file = filepath
+   } else {
+     .file = ".journald.system."
+   }
+   
+   if ( .host == null ) {
+      .host = .kubernetes.pod_ips
+   }
+
+   if (.GroupBy == "logType") {
+      .Cw_groupName =  ( .LogGroupPrefix + "." + .log_type ) ?? {}
+     if (.log_type == "application" ) {
+      .Cw_streamName =  .file
+     } else {
+      .Cw_streamName = ( .host + .file )  ?? {} 
+     }
+   } 
+
+   if ( .GroupBy == "namespaceName"  || .GroupBy == "namespaceUUID" ) {
+     if (.log_type == "application" ) {
+      .Cw_groupName =  ( .LogGroupPrefix + "." + .kubernetes.namespace_name ) ?? {}
+      .Cw_streamName = .file
+     } else {
+      .Cw_groupName =  ( .LogGroupPrefix + "." + .log_type ) ?? {}
+      .Cw_streamName = (  .host + .file )  ?? {} 
+     }
+   }
```

Comment:
this generated VRL code is very heavy at runtime, and it doesn't need to be.
The values of log group prefix is known at generation time, so it does not need to be set in the log record, only to be read again to construct another value in the record. All other VRLs use `fmt.Sprintf` to inject values known at conf generation time.